newdle/answers.py

```diff
@@ -0,0 +1,59 @@
+from collections import defaultdict
+
+from newdle.core.util import format_dt
+from newdle.models import Availability
+
+
+def validate_answers(newdle, participant, answers):
+    invalid = answers.keys() - set(newdle.timeslots)
+    if invalid:
+        return {
+            'code': 422,
+            'messages': {
+                'answers': {
+                    format_dt(key): {'key': ['Invalid timeslot']} for key in invalid
+                }
+            },
+        }
+
+    if newdle.limited_slots:
+        # 'ifneedbe' is not allowed for newdles with limited slots
+        if Availability.ifneedbe in answers.values():
+            return {
+                'code': 422,
+                'messages': {
+                    'answers': [
+                        "'if-need-be' is not allowed in newdles with limited slots"
+                    ]
+                },
+            }
+
+        # Must have at most one 'available'
+        counts = defaultdict(int)
+        for availability in answers.values():
+            counts[availability] += 1
+        if counts[Availability.available] > 1:
+            return {
+                'code': 422,
+                'messages': {'answers': ["At most one 'available' may be present"]},
+            }
+
+        # The slot must not be already taken
+        timeslot = next(
+            slot
+            for slot, availability in answers.items()
+            if availability == Availability.available
+        )
+        for p in newdle.participants:
+            if p.code == participant.code:
+                continue
+
+            if p.answers.get(timeslot) == Availability.available:
```

Comment:
`any(p.answers.get(timeslot) == Availability.available for p in newdle.participants if p.code != participant.code)`?